packages/astro/src/cli/check/index.ts

```diff
@@ -1,101 +1,360 @@
 /* eslint-disable no-console */
-import { AstroCheck, DiagnosticSeverity } from '@astrojs/language-server';
+import { AstroCheck, DiagnosticSeverity, GetDiagnosticsResult } from '@astrojs/language-server';
 import type { AstroSettings } from '../../@types/astro';
 import type { LogOptions } from '../../core/logger/core.js';
-
 import glob from 'fast-glob';
 import * as fs from 'fs';
 import { bold, dim, red, yellow } from 'kleur/colors';
 import { createRequire } from 'module';
 import ora from 'ora';
 import { fileURLToPath, pathToFileURL } from 'url';
 import { printDiagnostic } from './print.js';
+import type { Arguments as Flags } from 'yargs-parser';
+import { debug, info } from '../../core/logger/core.js';
+import type { SyncOptions, ProcessExit } from '../../core/sync';
+import fsMod from 'fs';
+import type { FSWatcher } from 'chokidar';
+import { join } from 'node:path';
 
-interface Result {
+type DiagnosticResult = {
 	errors: number;
-	// The language server cannot actually return any warnings at the moment, but we'll keep this here for future use
 	warnings: number;
 	hints: number;
-}
+};
 
-export async function check(settings: AstroSettings, { logging }: { logging: LogOptions }) {
-	console.log(bold('astro check'));
+type CheckPayload = {
+	/**
+	 * Flags passed via CLI
+	 */
+	flags: Flags;
 
-	const { syncCli } = await import('../../core/sync/index.js');
-	const syncRet = await syncCli(settings, { logging, fs });
-	// early exit on sync failure
-	if (syncRet === 1) return syncRet;
+	/**
+	 * Logging options
+	 */
+	logging: LogOptions;
+};
 
-	const root = settings.config.root;
+type CheckFlags = {
+	/**
+	 * Whether the `check` command should watch for `.astro` and report errors
+	 * @default {false}
+	 */
+	watch: boolean;
+};
+
+/**
+ *
+ * Types of response emitted by the checker
+ */
+export enum CheckResult {
+	/**
+	 * Operation finished without errors
+	 */
+	ExitWithSuccess,
+	/**
+	 * Operation finished with errors
+	 */
+	ExitWithError,
+	/**
+	 * The consumer should not terminate the operation
+	 */
+	Listen,
+}
+
+const ASTRO_GLOB_PATTERN = '**/*.astro';
 
-	const spinner = ora(` Getting diagnostics for Astro files in ${fileURLToPath(root)}…`).start();
+/**
+ * Checks `.astro` files for possible errors.
+ *
+ * If the `--check` flag is provided, the command runs indefinitely and provides diagnostics
+ * when `.astro` files are modified.
+ *
+ * Every time an astro files is modified, content collections are also generated.
+ *
+ * @param {AstroSettings} settings
+ * @param {CheckPayload} options
+ * @param {Flags} options.flags
+ * @param {LogOptions} options.logging
```

Comment:
I'm not sure about these JSDoc's, as they're duplicating TypeScript types and might cause a chance of it being outdated. I think we can remove it if we're not adding documentation for a specific param.


There's some other similar pattern in this file which I think we can remove.

Reply:
That's also my opinion on JSDoc, it barely improves the hover information unless you add descriptions to every property and often just ends up being a (often outdated) duplicate of the types. Not a fan

Reply:
This is more about hinting the correct information to the user, without navigating too much the "types":

Without
![Screenshot 2023-03-03 at 15 29 45](https://user-images.githubusercontent.com/602478/222760715-e89ff115-6651-4492-aad5-d4245d261c2c.png)

With 
![Screenshot 2023-03-03 at 15 30 05](https://user-images.githubusercontent.com/602478/222760751-e1cd3272-d07a-4f30-ad31-cf09dadce67d.png)

I'm OK with removing the JSDoc, but I believe only TypeScript types are not enough to help developers.